### PR TITLE
marker opacity for scatter plot viz

### PR DESCRIPTION
**Pull request by moontrip**
This will address https://github.com/VEuPathDB/web-eda/issues/1487. Relying on https://github.com/VEuPathDB/web-components/pull/419. This also works with overlay and facet variables: tested but no screenshot is attached. 

Note that currently, the step of opacity is set to be 0.1.

Here are some screenshots:
- a general marker with opacity of 0
 scatter-viz-opacity0

- a general marker with opacity of 1
scatter-viz-opacity1

- a gradient marker with opacity of 0
scatter-viz-gradient-opacity0

- a gradient marker with opacity of 1
scatter-viz-gradient-opacity1

### CODE REVIEW

File: src/lib/core/components/visualizations/implementations/ScatterplotVisualization.tsx [hunk 1]
```diff
@@ -2246,6 +2276,15 @@ function processInputData<T extends number | string>(
             seriesGradientColorscale?.length > 0
               ? markerSymbolGradient
               : markerSymbol(index),
+          // need to set marker.line for a transparent case (opacity != 1)
```
Comment (bobular) [inline src/lib/core/components/visualizations/implementations/ScatterplotVisualization.tsx:2279]: The `opacity: 0.7` a few lines above here is being applied to the final plot, so when the user chooses 1 on the slider, they actually only get 0.7.

The solution may just to remove this line.

Comment (moontrip) [reply src/lib/core/components/visualizations/implementations/ScatterplotVisualization.tsx:2279]: @bobular Thank you for your suggestion. I will remove it! 👍

**Review by bobular — CHANGES_REQUESTED** [review]
It looks/works great! The only things missing are
a) opacity should be stored in vizConfig, not local state (unless that was explicitly not in the spec)
b) opacity should be added to thumbnail dependencies

**Comment by moontrip** [comment]
> It looks/works great! The only things missing are a) opacity should be stored in vizConfig, not local state (unless that was explicitly not in the spec) b) opacity should be added to thumbnail dependencies

@bobular Thank you for your valuable suggestions! 👍 I missed those parts at Viz. Will work for that, probably tomorrow and after done, I will certainly use "re-review request" icon 😉

**Comment by moontrip** [comment]
> It looks/works great! The only things missing are a) opacity should be stored in vizConfig, not local state (unless that was explicitly not in the spec) b) opacity should be added to thumbnail dependencies

@bobular I have addressed your suggestions. Since the corresponding web-components PR was already merged, you can test this PR without using npm-pack-here. But please don't forget to update your local repo (git pull origin scatterplot-opacity & yarn install) 😉

**Review by bobular — CHANGES_REQUESTED** [review]

**Comment by bobular** [comment]
Also, would it be useful to consolidate `markerColorOpacity` and `markerOpacity` into one name?  Perhaps `markerBodyOpacity` or `markerCenterOpacity`?

File: src/lib/core/components/visualizations/implementations/ScatterplotVisualization.tsx [hunk 2]
```diff
@@ -1220,7 +1232,7 @@ function ScatterplotViz(props: VisualizationProps<Options>) {
       ? plotSpacingOptions
       : undefined,
     // need to define markerColorOpacity for faceted plot
-    markerColorOpacity: markerColorOpacity,
+    markerColorOpacity: vizConfig.markerOpacity,
```
[the remark this answers is not in the document]
Comment (moontrip) [reply src/lib/core/components/visualizations/implementations/ScatterplotVisualization.tsx:1235]: @bobular Ah, yes. You are absolutely right. I will start working on this PR after the other work you commented/suggested, including new spec such as two banners etc. 👍

**Comment by moontrip** [comment]
> Also, would it be useful to consolidate `markerColorOpacity` and `markerOpacity` into one name? Perhaps `markerBodyOpacity` or `markerCenterOpacity`?

@bobular thank you for your suggestion. Perhaps markerBodyOpacity may work? I will need to change it at web-components as well 😄

**Comment by bobular** [comment]
> > Also, would it be useful to consolidate `markerColorOpacity` and `markerOpacity` into one name? Perhaps `markerBodyOpacity` or `markerCenterOpacity`?
> 
> @bobular thank you for your suggestion. Perhaps markerBodyOpacity may work? I will need to change it at web-components as well 😄

Oh, that's a pain re. web-components.  Maybe find a consistent name on the EDA-side that means no changes over there?

**Comment by moontrip** [comment]
> markerBodyOpacity

@bobular well, the current prop name, markerColorOpacity, is defined in web-components so it should be changed in there anyway. It may not be a big deal, except you will have one additional review request 😄

**Comment by moontrip** [comment]
@bobular I have addressed your feedbacks. Since the change of the prop name required to make a change at web-components, this PR now depends on https://github.com/VEuPathDB/web-components/pull/433.

**Comment by moontrip** [comment]
@bobular update. Due to fixing a bug for marker opacity work at web-components, https://github.com/VEuPathDB/web-components/pull/433 is now merged owing to Jeremy. So from now one this PR does not have a dependency with web-components.

**Review by bobular — APPROVED** [review]
This looks good - I'm going to approve without testing because I'm having trouble getting my dev instance up, and will focus my testing on #1464 when that is sorted

**Comment by moontrip** [comment]
@bobular Thank you for your reviews 👍

**Pull request merged**